General repository cleanup

gcampax opened the pull request:
Remove all the cruft, and put all stuff in a sensible order.

@Mehrad0711 why on Earth are Bootleg pipelines calling into SPL?

gcampax committed:
Remove the old scripts

No longer useful

gcampax committed:
Move pipeline definitions to the top-level of the repo

And split them into multiple files, for readability.

gcampax committed:
Add requirements.txt

So LGTM can pick up the dependencies and analyze our code

Mehrad0711 commented:
eval's functionality is a subset of eval-spl's. If you insist, we can change the eval step in `ned_train_eval` to use eval and move `train_eval_bootleg` to spl folder. I need to use spl repo in some of my experiments.

Mehrad0711 commented:
Also, I wanted to do a round of cleanup on the argument default values, removing defaults for owner, project, experiment for pipelines like bootleg which others will use.

gcampax commented:
> eval's functionality is a subset of eval-spl's. If you insist, we can change the eval step in `ned_train_eval` to use eval and move `train_eval_bootleg` to spl folder. I need to use spl repo in some of my experiments.

I looked more closely now, all the SPL specific stuff in evaluate-spl should go into the SPL repo, and you should just use plain evaluate.yaml

s-jse commented on Dockerfile:
```diff
@@ -22,26 +29,64 @@ RUN git fetch && git checkout ${GENIENLP_VERSION} && pip3 install --use-feature=
+RUN useradd -ms /bin/bash genie-toolkit
+
 ARG THINGTALK_VERSION=master
+# npm *really* does not like running as root, and will misbehave badly when
```
Seems like we run into a lot of problems when using npm.

gcampax replied:
Yes, but yarn is worse: it's outright broken with git dependencies

gcampax replied:
In fact, because git dependencies work fine in npm, we can remove the explicit thingtalk npm link, and rely on package-lock.json alone.

Mehrad0711 committed:
Rename do_ner argument to use_bootleg

Mehrad0711 committed:
Update default arguments for train_step

Mehrad0711 committed:
Remove train_eval steps from bootleg pipeline

The training and evaluation can be taken care of using train_eval pipelines

Mehrad0711 committed:
Remove default value for eval_set

Let user provide the name to avoid mistakes

Mehrad0711 committed:
Update default arguments for "everything"

Mehrad0711 committed:
Drop empty args from final params too

Mehrad0711 committed:
Remove default task_name for general pipelines

gcampax committed:
Fix auto-annotate component

gcampax committed:
Fix SimpleTOD, SUMBT

And make the image a parameter of both pipelines, which makes it
easier to fix bugs later.

gcampax committed:
Fix broken aws sync calls in train

gcampax committed:
Add missing imports

gcampax committed:
Merge remote-tracking branch 'origin/wip/mehrad/kubeflow' into wip/cleanup

gcampax committed:
Remove dlg_side argument everywhere

For semantic parsing, the side is decided by the task. For
paraphrasing, it's decided by the paraphrase_subfolder argument.

gcampax committed:
Set default parameter values consistently where required

gcampax commented:
Ok, I tested the train and eval components with this, and the other pipelines compile without errors. I'm going to merge this.

@Mehrad0711 I have *not* uploaded any SPL or Bootleg pipeline. Please check them and/or upload fixes as needed. Also, please delete all the old pipelines that are no longer relevant given the separate Bootleg pipeline and merge of SPL into regular training.
